### PR TITLE
feat: implement database connection and create job service

### PR DESCRIPTION
**Pull request by JosuePrado**
# ID-06: Data Base connection with Node
[Click to go to the US](https://trello.com/c/KNw34NSF/8-id-06-data-base-connection-with-node)
---
## What did I do it?
- Implement graph ql to the project 
- Implement database connection to the project
- Creation of job module (service, create job resolver)
- Creation of models and entities for job

## How did I do it?
- The prisma orm framework was used for the connection with the database
- Integrating the graph ql library

## How to test
- Use node 18.18.0 or higher 
- run `npm install`
- lift database with the command `docker compose -f ./docker-compose.dev.yml up`
- run `npx prisma generate` 
- run `npm start`
- go to `http://localhost:3000/graphql` to test

### CODE REVIEW

**Review by alex-paca-jalafund — CHANGES_REQUESTED** [review]
/home/fundacion/Git-University-Projects/JobService/node_modules/@nestjs/graphql/dist/schema-builder/factories/output-type.factory.js:19
                throw new cannot_determine_output_type_error_1.CannotDetermineOutputTypeError(hostType);
                      ^

CannotDetermineOutputTypeError: Cannot determine a GraphQL output type for the "createJob". Make sure your class is decorated with an appropriate decorator.
    at OutputTypeFactory.create (/home/fundacion/Git-University-Projects/JobService/node_modules/@nestjs/graphql/dist/schema-builder/factories/output-type.factory.js:19:23)
    at /home/fundacion/Git-University-Projects/JobService/node_modules/@nestjs/graphql/dist/schema-builder/factories/root-type.factory.js:37:49
    at Array.forEach (<anonymous>)
    at RootTypeFactory.generateFields (/home/fundacion/Git-University-Projects/JobService/node_modules/@nestjs/graphql/dist/schema-builder/factories/root-type.factory.js:35:14)
    at fieldsFactory (/home/fundacion/Git-University-Projects/JobService/node_modules/@nestjs/graphql/dist/schema-builder/factories/root-type.factory.js:19:101)
    at RootTypeFactory.create (/home/fundacion/Git-University-Projects/JobService/node_modules/@nestjs/graphql/dist/schema-builder/factories/root-type.factory.js:28:21)
    at MutationTypeFactory.create (/home/fundacion/Git-University-Projects/JobService/node_modules/@nestjs/graphql/dist/schema-builder/factories/mutation-type.factory.js:15:37)
    at GraphQLSchemaFactory.create (/home/fundacion/Git-University-Projects/JobService/node_modules/@nestjs/graphql/dist/schema-builder/graphql-schema.factory.js:39:48)
    at GraphQLSchemaBuilder.generateSchema (/home/fundacion/Git-University-Projects/JobService/node_modules/@nestjs/graphql/dist/graphql-schema.builder.js:38:52)
    at GraphQLSchemaBuilder.build (/home/fundacion/Git-University-Projects/JobService/node_modules/@nestjs/graphql/dist/graphql-schema.builder.js:22:31)

File: src/services/jobs/dto/CreateJob.input.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  job_type: string;
+
+  @Field()
+  status: string;
+
+  @Field()
+  description: string;
+
+  @Field()
+  created_at: Date;
```
Comment (luiggy-mamani-jalafund) [inline src/services/jobs/dto/CreateJob.input.ts]: you don't need to request the created at attribute from the client, this should be generated from the server

File: src/services/jobs/dto/job.model.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { Field, Float, InputType } from '@nestjs/graphql';
+
+@InputType()
+export class jobModel {
```
Comment (luiggy-mamani-jalafund) [inline src/services/jobs/dto/job.model.ts:4]: this DTO looks similar to the `create job` DTO, so why do you need this DTO?

**Review by luiggy-mamani-jalafund — CHANGES_REQUESTED** [review]

File: src/services/jobs/dto/CreateJob.input.ts (hunk 1 above)
Comment (JosuePrado) [reply src/services/jobs/dto/CreateJob.input.ts]: You are right, i will fix it

File: src/services/jobs/dto/job.model.ts (hunk 2 above)
Comment (JosuePrado) [reply src/services/jobs/dto/job.model.ts:4]: I forgot to put the id field, which is the main difference, in this DTO we have the ID of the job, which is something we don't have in the CreateJob Input DTO

**Comment by JosuePrado** [comment]
> /home/fundacion/Git-University-Projects/JobService/node_modules/@nestjs/graphql/dist/schema-builder/factories/output-type.factory.js:19 throw new cannot_determine_output_type_error_1.CannotDetermineOutputTypeError(hostType); ^
> 
> CannotDetermineOutputTypeError: Cannot determine a GraphQL output type for the "createJob". Make sure your class is decorated with an appropriate decorator. at OutputTypeFactory.create (/home/fundacion/Git-University-Projects/JobService/node_modules/@nestjs/graphql/dist/schema-builder/factories/output-type.factory.js:19:23) at /home/fundacion/Git-University-Projects/JobService/node_modules/@nestjs/graphql/dist/schema-builder/factories/root-type.factory.js:37:49 at Array.forEach () at RootTypeFactory.generateFields (/home/fundacion/Git-University-Projects/JobService/node_modules/@nestjs/graphql/dist/schema-builder/factories/root-type.factory.js:35:14) at fieldsFactory (/home/fundacion/Git-University-Projects/JobService/node_modules/@nestjs/graphql/dist/schema-builder/factories/root-type.factory.js:19:101) at RootTypeFactory.create (/home/fundacion/Git-University-Projects/JobService/node_modules/@nestjs/graphql/dist/schema-builder/factories/root-type.factory.js:28:21) at MutationTypeFactory.create (/home/fundacion/Git-University-Projects/JobService/node_modules/@nestjs/graphql/dist/schema-builder/factories/mutation-type.factory.js:15:37) at GraphQLSchemaFactory.create (/home/fundacion/Git-University-Projects/JobService/node_modules/@nestjs/graphql/dist/schema-builder/graphql-schema.factory.js:39:48) at GraphQLSchemaBuilder.generateSchema (/home/fundacion/Git-University-Projects/JobService/node_modules/@nestjs/graphql/dist/graphql-schema.builder.js:38:52) at GraphQLSchemaBuilder.build (/home/fundacion/Git-University-Projects/JobService/node_modules/@nestjs/graphql/dist/graphql-schema.builder.js:22:31)

fixed

**Review by luiggy-mamani-jalafund — APPROVED** [review]

**Review by alex-paca-jalafund — APPROVED** [review]

**Pull request merged**